tests/testthat/test-knitr_formats.R

```diff
@@ -104,9 +104,59 @@ test_that("it does _not_ error with inline \\Sexpr", {
   )
 })
 
-test_that("it does error with malformed input", {
-  expect_error(
-    lint("knitr_malformed/incomplete_r_block.Rmd"),
-    regex = "Malformed"
+test_that("it does lint with malformed input", {
+  expect_lint(
+    file = "knitr_malformed/incomplete_r_block.Rmd",
+    checks = "Missing chunk end",
+    default_linters
+  )
+
+  contents <- c(
+    paste(
+      "```{r chunk}",
```

Comment:
WDYT about putting an ad hoc file into a string like this

    contents <- "```{r chunk}
    lm(x ~ y)
    
    
    # some text
    
    ```
    bash some_script.sh
    ```
    "

more readable than being cluttered with `"` IMO

Reply:
Wouldn't the indentation be messed up either in the source code or in the ad hoc file?
Maybe a helper function that automatically detects the correct amount of trimming would produce nice code and nice ad-hoc files.

Something like

```r
if (lala) {
  ad_hoc <- trim_some("
    line1
    line2
      line3 indented
  ")
}
```

Which should produce the same value as
`ad_hoc <- "line1\nline2\n__line3 indented"` (double underscore instead of space to prevent github trimming)

Reply:
The idea is for `trim_some` to trim relative to the first line? or I guess relative to the least-indented line. Looks good anyway, probably OK to just add such helper in `tests/testthat` only.

Reply:
Exactly, relative to the least indented line. I guess I might as well add this in the current PR since travis had a random hiccup preventing merge.